app/controllers/products_controller.rb

```diff
@@ -1,2 +1,75 @@
 class ProductsController < ApplicationController
+
+  # Add controller filter once set up
+  # before_action :find_product, only: [:show, :edit, :update, :destroy?]
+
+  def index
+    @products = Product.all
+  end
+
+  def new
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.new(product_params)
```

Comment:
current_merchant = find_by(merchant id.
if nil?
404 /need to login
otherwise
proceed...

assign id number to product as well. << take the local variable passenger = Passenger.find_by(id: session[:passenger_id] 
That should be logged in person.
You can do @product = Merchant.products.new & all parameters

Like creating a new trip in rideshare rails:
in #1 
merchant.products.new


to-do:
do migration for relation